priv/repo/migrations/20200316103722_create_user_auth_tables.exs

```diff
@@ -0,0 +1,26 @@
+defmodule Demo.Repo.Migrations.CreateUserAuthTables do
+  use Ecto.Migration
+
+  def change do
+    execute "CREATE EXTENSION IF NOT EXISTS citext", ""
+
+    create table(:users) do
+      add :email, :citext, null: false
+      add :hashed_password, :string, null: false
+      add :confirmed_at, :naive_datetime
+      timestamps()
+    end
+
+    create unique_index(:users, [:email])
+
+    create table(:user_tokens) do
+      add :user_id, references(:users, on_delete: :delete_all), null: false
+      add :token, :binary, null: false
```

Comment:
If you wan to use this pattern for say OAUTH style tokens you'll need to Base64 encode them before using them. Or change this to text and base64 encode before saving. In postgres there is limited valuer in using varchar or binary for such short binary values. Text is fine for almost everything these days.